src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.Unix.cs

```diff
@@ -39,7 +37,17 @@ public UnixDomainSocketEndPoint(string path)
             }
 
             _path = path;
-            _encodedPath = s_pathEncoding.GetBytes(_path);
+
+            // Pathname socket addresses should be null-terminated.
+            // Linux abstract socket addresses start with a zero byte, they must not be null-terminated.
+            bool isAbstract = path.Length > 0 && path[0] == '\0';
+            int length = s_pathEncoding.GetByteCount(path);
+            if (!isAbstract)
+            {
+                length++;
+            }
+            _encodedPath = new byte[length];
+            s_pathEncoding.GetBytes(path, 0, path.Length, _encodedPath, 0);
 
             if (path.Length == 0 || _encodedPath.Length > s_nativePathLength)
```

Comment:
We should move this check up to after we compute `length` and use that rather than `_encodedPath.Length`, before we allocate the array and do the actual encoding.

Reply:
The `path.Length == 0` check can go above all the new code.  That way `isAbstract` doesn't need to check `path.Length > 0` as well.